src/decompiler-ast.h

```diff
@@ -340,15 +355,21 @@ struct AST {
     if (size != 1) {
       InsertNode(NodeType::Statements, ExprType::Nop, nullptr, size);
     }
+    blocks_closed[cur_block_id] = true;
+    cur_block_id = block_stack.back();
+    block_stack.pop_back();
   }
 
   ModuleContext& mc;
   std::vector<Node> exp_stack;
   std::vector<Node> predecls;
   const Func *f;
   int value_stack_depth = 0;
-  std::set<std::string> vars_defined;
+  std::map<std::string, std::pair<size_t, bool>> vars_defined;
```

Comment:
Probably worth a comment what the pairs represent here. This is simple and scoped enough that a struct is probably overkill, but some clear reminder would be nice.

Reply:
Actually, maybe make it a struct, given that I already had an occurrence of `ret.first->second.first` :)
Structs are so lightweight in recent C++, `pair` is almost an anti-pattern.